src/managers/GuildMemberManager.js

```diff
@@ -224,35 +225,40 @@ class GuildMemberManager extends BaseManager {
       .then(data => this.add(data, cache));
   }
 
-  _fetchMany({ limit = 0, withPresences: presences = false, user: user_ids, query, time = 120e3 } = {}) {
+  _fetchMany({ limit = 0, withPresences: presences = false, user: user_ids, query, time = 120e3, nonce } = {}) {
     return new Promise((resolve, reject) => {
       if (this.guild.memberCount === this.cache.size && !query && !limit && !presences && !user_ids) {
         resolve(this.cache);
         return;
       }
       if (!query && !user_ids) query = '';
+      if (!nonce) nonce = this.guild.id;
       this.guild.shard.send({
         op: OPCodes.REQUEST_GUILD_MEMBERS,
         d: {
           guild_id: this.guild.id,
           presences,
           user_ids,
           query,
+          nonce,
           limit,
         },
       });
       const fetchedMembers = new Collection();
       const option = query || limit || presences || user_ids;
-      const handler = (members, guild) => {
-        if (guild.id !== this.guild.id) return;
+      let i = 0;
+      const handler = (members, _, chunk) => {
+        if (chunk.nonce !== nonce) return;
```

Comment:
Could we refresh the timeout even if we don't get a chunk with the same nonce? That way, if multiple requests have been done, (across even multiple guilds), there's no chance for an expiry just cause we're waiting on the specific packet to arrive

Reply:
I'm not sure I get your point, if two requests are being sent there will be one listener for each of them with their respective timeout 🤔.
If a packet do not arrive in time for request A, it shouldn't prevent request B to complete.

Reply:
This won't be an issue for two requests, but think MANY requests over, say, 1k guilds, each with a LOT of members to return (remember: a packet can have a max of 1k members). Assuming all of them were requested sequentially, guild 1000 could timeout waiting for packets while the other handlers.

Reply:
I didn't think about this case actually, this is a good catch. Latest commit moves `timeout.refresh();` above nonce's check.